packages/ember-metal/lib/mixin.js

```diff
@@ -633,9 +633,10 @@ function _keys(ret, mixin, seen) {
   seen[guidFor(mixin)] = true;
 
   if (mixin.properties) {
-    var props = mixin.properties;
-    for (var key in props) {
-      if (props.hasOwnProperty(key)) { ret[key] = true; }
```

Comment:
I just thought of it while reading, and thought maybe we should have a quick discussion.